### PR TITLE
update: `cross-chain-messaging counter`

### PR DESCRIPTION
**Pull request by 1337-ishaan**
Included steps to write `cross-chain-messaging counter`

### CODE REVIEW

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md [hunk 1]
```diff
@@ -27,8 +27,140 @@ yarn add --dev @openzeppelin/contracts
...
+                destinationChainId: destinationChainId,
+                destinationAddress: interactorsByChainId[destinationChainId],
+                destinationGasLimit: 300000,
+                message: abi.encode(CROSS_CHAIN_COUNTER_MESSAGE_TYPE, sender),
+                zetaValueAndGas: zetaValueAndGas,
+                zetaParams: abi.encode("")
+            })
+        );
+    }
+    //remove-end
```
Comment (fadeev) [inline docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:101]: I don't see a reason to remove the whole function just to then add it back. I believe we only need to remove `address sender` from the arg list, let's keep the `sendMessage` function name for consistency and update the `sender` to `msg.sender`.

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md [hunk 2]
```diff
@@ -27,8 +27,140 @@ yarn add --dev @openzeppelin/contracts
 
 ## Create a new contract
 
-```solidity title="contracts/CrossChainCounter.sol" reference
-https://github.com/zeta-chain/example-contracts/blob/3e79d006239bb9cbe0120e36aecc471cea4a2ad7/messaging/counter/contracts/CrossChainCounter.sol
+```
+npx hardhat messaging CrossChainCounter messageFrom:address
```
Comment (fadeev) [inline docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:31]:
```suggestion
npx hardhat messaging CrossChainCounter from:address
```

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md [hunk 3]
```diff
@@ -27,8 +27,140 @@ yarn add --dev @openzeppelin/contracts
...
+    error DecrementOverflow();
+}
+
+contract CrossChainCounter is
+    ZetaInteractor,
+    ZetaReceiver,
+    CrossChainCounterErrors
+{
+    // highlight-start
+    bytes32 public constant CROSS_CHAIN_INCREMENT_MESSAGE =
```
Comment (fadeev) [inline docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:58]: Let's make as few changes as possible to streamline the developer experience. There is only one message type, we might as well keep it the way it was generated by the `messaging` command: `CROSS_CHAIN_COUNTER_MESSAGE_TYPE`.

**Review by fadeev — CHANGES_REQUESTED** [review]
Thanks for tackling this issue!

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md (hunk 1 above)
Comment (fadeev) [reply docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:101]: One more reason to keep `sendMessage` is that it won't require chaning the `interact` task.

**Comment by fadeev** [comment]
We probably want to remove the section about deploy task, because it's now generated automatically and update the section on the interact task, because we no longer need to create this task from scratch, but we need to update it (remove the from argument generated by default).

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md (hunk 1 above)
Comment (1337-ishaan) [reply docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:101]: Let me get this done really quick!

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md (hunk 2 above)
Comment (1337-ishaan) [reply docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:31]: Noted sir!

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md [hunk 4]
```diff
@@ -27,19 +27,152 @@ yarn add --dev @openzeppelin/contracts
...
+        address connectorAddress,
+        address zetaTokenAddress,
+        address zetaConsumerAddress
+    ) ZetaInteractor(connectorAddress) {
+        _zetaToken = IERC20(zetaTokenAddress);
+        _zetaConsumer = ZetaTokenConsumer(zetaConsumerAddress);
+    }
+
+
+    function sendMessage(uint256 destinationChainId, address sender) external payable {
```
Comment (fadeev) [inline docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:80]:
```suggestion
    function sendMessage(uint256 destinationChainId) external payable {
```

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md [hunk 5]
```diff
@@ -27,19 +27,152 @@ yarn add --dev @openzeppelin/contracts
...
+                message: abi.encode(CROSS_CHAIN_COUNTER_MESSAGE_TYPE, msg.sender), // sender --> msg.sender
+                zetaValueAndGas: zetaValueAndGas,
+                zetaParams: abi.encode("")
+            })
+        );
+    }
+
+
+    // highlight-start
+    function crossChainCount(uint256 destinationChainId) external payable {
```
Comment (fadeev) [inline docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:105]: Please, remove this function. We already have `sendMessage` generated by default.

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md (hunk 3 above)
Comment (fadeev) [reply docs/developers/cross-chain-messaging/examples/cross-chain-counter.md]: Ok, I didn't realize it generates `CROSS_CHAIN_CROSS_CHAIN_COUNTER`, which is not pretty. We might want to change the template in the future to be less verbose, but for now I think it makes sense to just switch to:

```
npx hardhat messaging counter from:address
```

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md (hunk 5 above)
Comment (1337-ishaan) [reply docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:105]: Noted!

**Review by fadeev — APPROVED** [review]
Looks good! Thanks for your contribution!

**Pull request merged**